File: 3_Examples/sample.cpp
#include <cudaq.h>
#include <cudaq/algorithms/draw.h>
#include <iostream>

__qpu__ void kernal(const int qubitCount)
{

    auto qvector = cudaq::qvector(qubitCount);

    cudaq::h(qvector[0]);

    for (int i : cudaq::range(1, qubitCount))
    {
        cudaq::x<cudaq::ctrl>(qvector[0], qvector[1]);
    }
}

int main(int argc, char const *argv[])
{
    int qubitCount = 2;
    std::cout << cudaq::draw(kernal, qubitCount);

    auto result = cudaq::sample(1000, kernal, qubitCount);

    result.dump();

    return 0;
}

File: 3_Examples/sample.py
import cudaq
import numpy as np

qubit_count = 2

# Define the simulation target.
cudaq.set_target("qpp-cpu")

# Define a quantum kernel function.


@cudaq.kernel
def kernel(qubit_count: int):
    qvector = cudaq.qvector(qubit_count)

    # 2-qubit GHZ state.
    h(qvector[0])
    for i in range(1, qubit_count):
        x.ctrl(qvector[0], qvector[i])

    # If we dont specify measurements, all qubits are measured in
    # the Z-basis by default or we can manually specify it also
    # mz(qvector)


print(cudaq.draw(kernel, qubit_count))

result = cudaq.sample(kernel, qubit_count, shots_count=1000)

print(result)
